tests/returntypes/test_parameter_shift_hessian_new.py

```diff
@@ -0,0 +1,1849 @@
+    def test_all_zero_grads(self, num_measurements):
+        """Test that the transform works correctly when the diff method for every parameter is
+        identified to be 0, and that no tapes were generated."""
+        dev = qml.device("default.qubit", wires=2)
+
+        class DummyOp(qml.CRZ):
+            grad_method = "0"
+
+        x = np.array(0.1, requires_grad=True)
+
+        with qml.tape.QuantumTape() as tape:
+            DummyOp(x, wires=[0, 1])
+            for _ in range(num_measurements):
+                qml.probs(wires=[0, 1])
+
+        tapes, fn = qml.gradients.param_shift_hessian(tape)
+        res = fn(qml.execute(tapes, dev, None))
+
+        if num_measurements == 1:
+            res = (res,)
+
+        assert tapes == []
+        assert isinstance(res, tuple)
+        assert len(res) == num_measurements
+        assert all(
+            isinstance(r, np.ndarray) and np.allclose(r, np.array([0, 0, 0, 0])) for r in res
+        )
+
+    def test_error_unsupported_op(self):
+        """Test that the correct error is thrown for unsupported operations"""
+
+        dev = qml.device("default.qubit", wires=2)
+
+        class DummyOp(qml.CRZ):
+            grad_method = "F"
+
+        x = np.array([0.1, 0.2, 0.3], requires_grad=True)
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(x[0], wires=0)
+            qml.RY(x[1], wires=0)
+            DummyOp(x[2], wires=[0, 1])
+            qml.probs(wires=1)
+
+        msg = "The parameter-shift Hessian currently does not support the operations"
+        with pytest.raises(ValueError, match=msg):
+            qml.gradients.param_shift_hessian(tape, argnum=[0, 1, 2])(x)
+
+    @pytest.mark.parametrize("argnum", [None, (0,)])
+    def test_error_wrong_diagonal_shifts(self, argnum):
+        """Test that an error is raised if the number of diagonal shifts does
+        not match the required number (`len(trainable_params)` or `len(argnum)`)."""
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(0.4, wires=0)
+            qml.CRY(0.9, wires=[0, 1])
+
+        with pytest.raises(ValueError, match="sets of shift values for diagonal entries"):
+            qml.gradients.param_shift_hessian(tape, argnum=argnum, diagonal_shifts=[])
+
+    @pytest.mark.parametrize("argnum", [None, (0, 1)])
+    def test_error_wrong_offdiagonal_shifts(self, argnum):
+        """Test that an error is raised if the number of offdiagonal shifts does
+        not match the required number (`len(trainable_params)` or `len(argnum)`)."""
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(0.4, wires=0)
+            qml.CRY(0.9, wires=[0, 1])
+            qml.RX(-0.4, wires=0)
+
+        with pytest.raises(ValueError, match="sets of shift values for off-diagonal entries"):
+            qml.gradients.param_shift_hessian(tape, argnum=argnum, off_diagonal_shifts=[])
+
+
+@pytest.mark.skip("Requires QNode integration for new return types")
```

Comment:
You should be able to add those tests, maybe worth to wait for Jax PR. As the return structure should be similar.